File: trunk/theLink/example/ruby/Filter1.rb
#+
#§  \file       theLink/example/ruby/Filter1.rb
#§  \brief      \$Id: LbMain 26 2009-12-03 11:48:43Z aotto1968 $
#§  
#§  (C) 2010 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev: 26 $
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

require "rubymsgque"

class Filter1 < MqS
  def initialize
    ConfigSetName("filter")
    ConfigSetServerSetup(method(:ServerSetup))
    @data = []
    super()
  end
  def ServerSetup
    ServiceCreate("+FTR", method(:FTRcmd))
    ServiceCreate("+EOF", method(:EOFcmd))
  end
  def FTRcmd
    list = []
    while ReadItemExists()
      list << "<" + ReadC() + ">"
    end
    @data << list
    SendRETURN()
  end
  def EOFcmd
    ftr = ServiceGetFilter()
    for list in @data
      ftr.SendSTART()
      for item in list
        ftr.SendC(item)
      end
      ftr.SendEND_AND_WAIT("+FTR")
    end
    ftr.SendSTART()
    ftr.SendEND_AND_WAIT("+EOF")
    SendRETURN()
  end
end
srv = Filter1.new
begin
  srv.LinkCreate($0,ARGV)
  srv.ProcessEvent(MqS::WAIT_FOREVER)
rescue SignalException => ex
  # ignore
rescue Exception => ex
  srv.ErrorSet(ex)
ensure
  srv.Exit()
end

File: trunk/theLink/example/php/Filter1.php
<?php
#+
#:  \file       theLink/example/php/Filter1.php
#:  \brief      \$Id$
#:  
#:  (C) 2010 - NHI - #1 - Project - Group
#:  
#:  \version    \$Rev$
#:  \author     EMail: aotto1968 at users.berlios.de
#:  \attention  this software has GPL permissions to copy
#:              please contact AUTHORS for additional information
#:

class Filter1 extends MqS implements iServerSetup {
  public function __construct($tmpl=NULL) {
    parent::__construct($tmpl);
    $this->data = array();
  }
  public function ServerSetup() {
    $this->ServiceCreate('+FTR', array(&$this, 'FTRcmd'));
    $this->ServiceCreate('+EOF', array(&$this, 'EOFcmd'));
  }
  public function FTRcmd() {
    $list = array();
    while ($this->ReadItemExists()) {
      $item = $this->ReadC();
      $list[] = "<$item>";
    }
    $this->data[] = $list;
    $this->SendRETURN();
  }
  public function EOFcmd() {
    $ftr = $this->ServiceGetFilter();
    foreach ($this->data as $list) {
      $ftr->SendSTART();
      foreach ($list as $item) {
        $ftr->SendC($item);
      }
      $ftr->SendEND_AND_WAIT('+FTR');
    }
    $ftr->SendSTART();
    $ftr->SendEND_AND_WAIT('+EOF');
    $this->SendRETURN();
  }
}
MqS::Init(PHP_BINARY, "-c", get_cfg_var('cfg_file_path'), __FILE__);
$srv = FactoryAdd('filter', 'Filter1')->New();
try {
  $srv->LinkCreate($argv);
  $srv->ProcessEvent(MqS::WAIT_FOREVER);
} catch (Exception $ex) {
  $srv->ErrorSet($ex);
}
$srv->Exit();

?>

